src/main/shell-session/node-shell-session.ts

```diff
@@ -55,10 +57,27 @@ export class NodeShellSession extends ShellSession {
       throw new ShellOpenError("failed to create node pod", error);
     }
 
-    const args = ["exec", "-i", "-t", "-n", "kube-system", this.podId, "--", "sh", "-c", "((clear && bash) || (clear && ash) || (clear && sh))"];
     const env = await this.getCachedShellEnv();
+    const args = ["exec", "-i", "-t", "-n", "kube-system", this.podId, "--"];
+    const nodeApi = new NodesApi({
+      objectConstructor: Node,
+      request: KubeJsonApi.forCluster(this.cluster),
+    });
+    const node = await nodeApi.get({ name: this.nodeName });
+    const nodeOs = node.getOperatingSystem();
+
+    switch (nodeOs) {
+      case "linux":
+        args.push("sh", "-c", "((clear && bash) || (clear && ash) || (clear && sh))");
+        break;
+      case "windows":
+        args.push("powershell");
+        break;
+      default:
+        throw new Error(`Cannot open node shell for ${this.nodeName}: unknown operating sytem ${nodeOs}`);
```

Comment:
nit: previous behaviour just tried linux style regardless, you could just log a message here and proceed as linux